lib/bolt/target.rb

```diff
@@ -47,8 +47,8 @@ def update_conf(conf)
       @password = t_conf['password']
       @port = t_conf['port']
 
-      url_keys = %w[user password port]
-      @options = t_conf.reject { |k, _| url_keys.include?(k) }.merge(@options)
+      # Preserve everything in options so we can easily create copies of a Target.
```

Comment:
This was necessary to make the Local transport play nice when changing tmpdir. I needed to create a copy of the target without tmpdir. I think the change makes sense on its own.